presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -198,6 +198,18 @@ public synchronized void load()
             configureChangedGroups(changedSpecs);
             disableDeletedGroups(deletedSpecs);
 
+            if (lastRefresh.get() > 0) {
+                for (ResourceGroupIdTemplate deleted : deletedSpecs) {
+                    log.info("Resource group spec deleted %s", deleted);
+                }
+                for (ResourceGroupIdTemplate changed : changedSpecs) {
+                    log.info("Resource group spec % changed to %s", changed, resourceGroupSpecs.get(changed));
+                }
+            }
+            else {
+                log.info("Loaded %s selectors and %s resource groups from database", this.selectors.get().size(), this.resourceGroupSpecs.size());
```

Comment:
both `%s` -> `%d`

Reply:
We always use %s because it's less likely to blow up.

Reply:
This is not a big issue, so I am fine either way. But, in this case there is no possibility of blow up as these are obviously integers. Also in general I don't think it's good practice to always use `%s` as `%d` will do locale-sensitive formatting (see [this](https://stackoverflow.com/questions/7872204/s-in-string-format-for-numbers/7872252#7872252) SO answer).

Reply:
Our preferred style is to use `%s` unless specific formatting is required. This is important since logging or error formatting often happens on code paths which aren't tested (not ideal but that's the reality). It also makes the code easier to read and review, since you don't need to think about and match up the formatting arguments.

The logging system is actually safe against invalid format strings, but the messages will look bad.

I wasn't aware that `%s` is locale-sensitive. That has some interesting implications. Probably fine for logging. Using the server locale for client facing errors is less than ideal. I suppose using `format(null, "%s ...", ...)` everywhere would be the right way.

Reply:
Oh, I see that it's `%d` which is locale-sensitive. I'd argue that we don't want localization for numbers, since we aren't localizing anything else, and getting it implicitly is very counter-intuitive.

However, it seems that neither one is localized in Java 9.